test/unit/Escrow.t.sol

```diff
@@ -1388,6 +1388,69 @@ contract EscrowUnitTests is UnitTest {
         );
     }
 
+    function test_withdrawETH_2_HappyPath_HolderPerspective() external {
+        uint256 balanceBefore = _vetoer.balance;
+        uint256[] memory unstEthAmounts = new uint256[](2);
+        unstEthAmounts[0] = _stETH.getPooledEthByShares(1 ether);
+        unstEthAmounts[1] = _stETH.getPooledEthByShares(10 ether);
+        uint256 totalSharesToBeWithdrawn = _stETH.getSharesByPooledEth(unstEthAmounts[0] + unstEthAmounts[1]);
+
+        uint256[] memory unstEthIds = _vetoerLockedUnstEth(unstEthAmounts);
+        uint256[] memory hints = _finalizeUnstEth(unstEthAmounts, unstEthIds);
+
+        _transitToRageQuit();
+        _ensureWithdrawalsBatchesQueueClosed();
+        _ensureRageQuitExtensionPeriodStartedNow();
+        assertTrue(_escrow.getRageQuitEscrowDetails().isRageQuitExtensionPeriodStarted);
+        _wait(Durations.from(1));
+
+        _claimUnstEthFromEscrow(unstEthAmounts, unstEthIds, hints);
+
+        IEscrow.VetoerDetails memory state = _escrow.getVetoerDetails(_vetoer);
+
+        assertApproxEqAbs(totalSharesToBeWithdrawn, state.unstETHLockedShares.toUint256(), ACCURACY);
+        assertEq(unstEthAmounts.length, state.unstETHIdsCount);
+
+        uint256[] memory firstWithdrawal = new uint256[](1);
+        firstWithdrawal[0] = unstEthIds[0];
+
+        vm.startPrank(_vetoer);
+        _escrow.withdrawETH(firstWithdrawal);
+        vm.stopPrank();
+
+        IEscrow.VetoerDetails memory stateAfterFirstWithdrawal = _escrow.getVetoerDetails(_vetoer);
+
+        // Based on current design it is expected that after the first withdrawal these data was not updated
+        assertApproxEqAbs(totalSharesToBeWithdrawn, stateAfterFirstWithdrawal.unstETHLockedShares.toUint256(), ACCURACY);
+        assertEq(unstEthAmounts.length, stateAfterFirstWithdrawal.unstETHIdsCount);
+        // But request is successful and funds are transferred
+        assertEq(balanceBefore + unstEthAmounts[0], _vetoer.balance);
+
+        uint256[] memory vetoerUnstETHIds = _escrow.getVetoerUnstETHIds(_vetoer);
+        IEscrow.LockedUnstETHDetails[] memory lockedUnstETHDetails = _escrow.getLockedUnstETHDetails(vetoerUnstETHIds);
+
+        uint256 index = 0;
+        for (uint256 i = 0; i < lockedUnstETHDetails.length; i++) {
+            if (lockedUnstETHDetails[i].status == UnstETHRecordStatus.Claimed) {
+                index++;
+            }
+        }
```

Comment:
This part of the code is unclear because of the misleading variable naming. It's better to rename the `index` variable into the `unclaimedUnstETHRecordsCount` or something.